include/boost/math/special_functions/detail/bessel_y0.hpp

```diff
@@ -145,6 +145,8 @@ T bessel_y0(T x, const Policy&)
 
     static const char* function = "boost::math::bessel_y0<%1%>(%1%,%1%)";
 
+    static_cast<void>(function[0U]);
```

Comment:
This can just be removed I think?

Reply:
I'm ok with syntax stuff. I tend to over-cast and handle warnings like a maniac. And thus (as some would say) obfuscate. I'll get back to this later after we hash out these basics.